### PR TITLE
Add a test of binding patterns

### PR DESCRIPTION
**Pull request by msridhar**
They seem to work correctly now; add a test to keep them working

### CODE REVIEW

File: jdk17-unit-tests/src/test/java/com/uber/nullaway/jdk17/NullAwayInstanceOfBindingTests.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  @Test
+  public void testInstanceOfBinding() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "InstanceOfBinding.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "class InstanceOfBinding {",
+            "  public void testInstanceOfBinding(@Nullable Object o) {",
+            "    if (o instanceof String s) {",
```
Comment (lazaroclapp) [inline jdk17-unit-tests/src/test/java/com/uber/nullaway/jdk17/NullAwayInstanceOfBindingTests.java:37]: Are the semantics here that `null instanceof Foo` is always `false`?

**Review by lazaroclapp — APPROVED** [review]
Nice! (Are we just getting this from CFG dataflow for free? Or where did we introduce support?)

**Comment by msridhar** [comment]
We didn't introduce any support; it just comes for free 🙂

File: jdk17-unit-tests/src/test/java/com/uber/nullaway/jdk17/NullAwayInstanceOfBindingTests.java (hunk 1 above)
Comment (msridhar) [reply jdk17-unit-tests/src/test/java/com/uber/nullaway/jdk17/NullAwayInstanceOfBindingTests.java:37]: Sorry missed this comment, yes that is correct

**Pull request merged**